### ISSUE TITLE
HOOBS randomly stops responding

### ISSUE THREAD

**Issue by the-icelighter** [hoobs-org/HOOBS#1256]
**Description**
My HOOBS in a box stops responding for no obvious reason. Connected accessories stop responding in HomeKit, I cannot login to the HOOBS UI and the HOOBS box no longer appears in the attached devices list of my router UI. The only solution is to remove and reconnect power, which resolves the issue but for an unpredictable amount of time. During the period of unresponsiveness I cannot retrieve the log. Forcing the unit to reboot by disconnecting power clears the log.

**Version**
Please include the version of HOOBS you are using.
3.3.5

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [x] Yes
- [ ] No

**Pervious version**
If you upgraded, please let us know your previous version.
When the HOOBS box was first booted I was prompted to update. I am unsure of the original version, but I'm fairly confident it was 3.x.x

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [ ] Yes
- [x] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [x] HOOBS Box
- [ ] HOOBS SD Card
- [ ] Custom Install

**If this is a custom install, what's your operating system?**
Please let us know what operating system and version you are using. Ex. Fedora 30 or macOS Catalina.
N/A

**List your plugins**
Please include a list of the plugins you are using. We need to be able to find it on NPM. If the plugin is no published, please include a link to the repository.
Ring - HOOBS Certified

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
{
    "server": {
        "port": 80,
        "origin": "*",
        "autostart": 5,
        "home_setup_id": "X-HM://0023ISYWYYB1K",
        "polling_seconds": 30
    },
    "client": {
        "default_route": "status",
        "inactive_logoff": 60,
        "theme": "hoobs-dark",
        "locale": "en",
        "temp_units": "celsius",
        "country_code": "GB",
        "postal_code": "REDACTED"
    },
    "bridge": {
        "name": "HOOBS",
        "port": 51826,
        "pin": "031-45-154",
        "username": "91:C9:64:37:52:AA"
    },
    "description": "",
    "ports": {},
    "accessories": [],
    "platforms": [
        {
            "platform": "Ring",
            "plugin_map": {
                "plugin_name": "homebridge-ring"
            },
            "refreshToken": "REDACTED",
            "alarmOnEntryDelay": false,
            "hideLightGroups": true,
            "hideDoorbellSwitch": true,
            "avoidSnapshotBatteryDrain": false,
            "hideCameraMotionSensor": false,
            "hideCameraSirenSwitch": false,
            "hideInHomeDoorbellSwitch": true,
            "hideAlarmSirenSwitch": true,
            "hideUnsupportedServices": true,
            "showPanicButtons": false,
            "sendCameraMotionNotificationsToTv": false,
            "sendDoorbellMotionNotificationsToTv": false,
            "debug": false,
            "cameraStatusPollingSeconds": 10,
            "cameraDingsPollingSeconds": 1,
            "locationModePollingSeconds": 0
        }
    ]
}
```

**Post your log**
You can get the log from the interface. This is in the Log section.

```
10/05/2021, 15:17:21 HOOBS listening on port 80.
10/05/2021, 15:17:28 Initializing HAP-NodeJS v0.9.3...
10/05/2021, 15:17:32 Loaded plugin 'homebridge-ring'
10/05/2021, 15:17:33 [5/10/2021, 8:17:32 AM] Registering platform 'homebridge-ring.Ring'
10/05/2021, 15:17:33 Loading 1 platforms...
10/05/2021, 15:17:33 [Ring] Initializing Ring platform...
10/05/2021, 15:17:33 [Ring] Configuring cached accessory 4234b9e4-90be-4e2e-b6de-34f2959a877b Front Door
10/05/2021, 15:17:33 [Ring] Configuring cached accessory af6c6ea5-a81c-402f-9dcf-261dbef058a8 Downstairs
10/05/2021, 15:17:33 Preparing Advertiser for 'HOOBS C20F' using bonjour-hap backend!
10/05/2021, 15:17:33 Starting to advertise 'HOOBS C20F' using bonjour-hap backend!
10/05/2021, 15:17:33 Bridge is running on port 51826.
10/05/2021, 15:17:36 [Ring] Found the following locations:
10/05/2021, 15:17:36 [Ring]   locationId: 0e2bf7c6-0287-45f2-6c25-ac6b6e4089a8 - Home
10/05/2021, 15:17:36 [Ring] Configuring 1 cameras and 2 devices for location "Home" - locationId: 0e2bf7c6-0287-45f2-6c25-ac6b6e4089a8
```

**Additional context**

**Comment by EddieDSuza** [comment]
The unresponsiveness happens when the box is assigned a LAN and a Wifi IP address. 

The logs you posted confirm there is no issue with the plugins but the connectivity of the box to the internet.

To resolve the issue - please SSH into the box. 

On MacOS use the Terminal app and type: ssh hoobs@hoobs.local

Type yes to accept the certificate key then press enter. When prompted for password, type hoobsadmin

On Windows, use Putty with the same credentials as above.

2. After successfully logging in, can you check the IP address allocated to the BOX.

If there are more than one - **reserve one of the IP address in the router and restart your network.**

This will fix the connectivity issue you are having.

**Comment by the-icelighter** [comment]
Thanks for the reply.

There are indeed two IP addresses. I already reserved one and have restarted my network several times (I restart on a regular basis). One is Ethernet and one is WiFi. I had my box wired up, but when this issue started I thought I'd try WiFi as well.

Is there a way to remove the WiFi credentials from the box? I've blocked the WiFi IP address on my router but it would probably be better if the box just didn't connect to the WiFi at all.

**Comment by EddieDSuza** [comment]
Well - when you reserved the IP - did it belong to the LAN or WIFI MAC address - please check. 

The only way to block the Wifi Credentials is to reserve the correct IP address to the MAC address of the LAN Network Interface card and not the Wifi Interface Card.

Even resetting or reflashing the IP address will not fix the issue because your router has mapped the BOX`s MAC address to its table.

**Comment by the-icelighter** [comment]
I've reserved the LAN MAC address.

I originally tried to reserve the WiFi address and unplugged the Ethernet, but the issue remained. So I moved my box to my main router (I have a mesh network), removed the WiFi reservation, checked the MAC addresses and reserved the Ethernet MAC address, then blocked the WiFi address. The issue persists.

I'll try reserving a different IP to the LAN MAC address, perform a network restart and wait for the results.

**Comment by EddieDSuza** [comment]
when using a Mesh Network - Always connect the BOX to the main node (this could be also another reason for failure to connect)

**Comment by mkellsy** [comment]
The log only contains the current run (changing in HOOBS 4)
To get the cause of why it restarted, SSH into the box and run this
```
sudo hoobs service log
```

It will show you the log before `10/05/2021, 15:17:21 HOOBS listening on port 80.`

**Comment by the-icelighter** [comment]
So, after reserving a different IP for the LAN MAC, the HOOBS box proved more reliable, staying connected for over 24 hours. However, it once again stopped responding and required a power cycle.

I tried SSH to retrieve logs prior to the power cycle and I can still only see logs after the box was booted.

**Comment by the-icelighter** [comment]
New update. I changed the WiFi SSID and password to prevent the HOOBS box from connecting wirelessly, as well as helping with general security.

The HOOBS box continues to stop responding.

I removed the IP reservation to see if the HOOBS box would use a different IP if it disconnected. No difference.

I'm at a loss. The HOOBS box is proving itself to be unreliable. Every other device connects and maintains connections flawlessly. I'm starting to consider the HOOBS box to be unfit for purpose.

If I cannot find respite in the community, I'll contact HOOBS directly.

**Comment by EddieDSuza** [comment]
Email info@hoobs.org and they will schedule a remote session to look into the issue.

**Comment by the-icelighter** [comment]
Found these entries in the log. Are they significant?


[15/05/2021, 02:01] Got SIGINT, shutting down Bridge...

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] [Refresh Services] connect ECONNREFUSED 127.0.0.1:51826

[15/05/2021, 02:02] Initializing HAP-NodeJS v0.9.3...

**Comment by EddieDSuza** [comment]
Can you paste the entire log or you can attach the file as well. Go to help > back > logs 

The content you pasted above is not complete - we need to see the information before the line below 

[15/05/2021, 02:01] Got SIGINT, shutting down Bridge...

**Comment by the-icelighter** [comment]
Sorry, I can't retrieve that log anymore. Further power cycles were required.

However, I believe I may have found the culprit.

I decided to attach my TV to the HOOBS box using HDMI. I found an under voltage warning on the displayed UI. I tried a more powerful plug than the included one; more specifically an Apple 15W USB-C power supply. I got the same result, so I tried a 20W variant of the Apple USB-C. Same again, under voltage. So I tried the 20W supply with a different USB cable. This resulted in no under voltage warning. Perhaps I was supplied with an inadequate USB cable?

I'll monitor the results and post back here.

Additionally, when I connected the 15W supply the red LED on the Raspberry Pi board didn't light up at first, but did after around 30 seconds. Is the red LED supposed to light during normal operation?

**Comment by EddieDSuza** [comment]
The red light is normal. Well from your comment the culprit is the USB cable. 

Can you use this cable and connect it with the power brick that came along with the box? 

Additionally are you able to restore your HOOBS system ?

**Comment by mkellsy** [comment]
It sounds like a bad board. It happens. Send an email to info@hoobs.org with your order number and a description. We can start the RMA process.

**Comment by the-icelighter** [comment]
Just to keep everyone who's been helping up to date, I've performed further troubleshooting that seems to suggest that mkellsy is correct; the board is faulty.

Since HOOBS v4 keeps logs prior to the current run I decided to try the v4 beta. This would also eliminate software as an issue, as well as the SD card, as I used a different one for the beta.

The issue persists on v4 with the different SD card. Nothing untoward appears in the log when the unresponsive occurs. In fact, no logs appear for the duration of the outage.

Additionally, when running the HOOBS box (v3 or v4) connected to HDMI, which requires the box to be open, the issue doesn't seem to occur. Perhaps the processor is reaching too high temperatures? Normal operation see temperatures of around 45°C to 48°C, which I wouldn't consider too hot, but I'm no expert. Also, the ventilation holes in the HOOBS box didn't seem completely machined, so I had to widen them with a pin. Could this have contributed to the problem?

Regardless, it seems the issue is to do with hardware. I'll email info@hoobs.org and request the RMA soon.
